lib/galaxy/datatypes/text.py

```diff
@@ -726,3 +727,37 @@ def sniff_prefix(self, file_prefix):
         False
         """
         return file_prefix.startswith("IQ-TREE")
+
+
+@build_sniff_from_prefix
+class Yaml(Text):
+    """Yaml files"""
+    file_ext = "yaml"
+
+    def sniff_prefix(self, file_prefix):
+        """
+            Try to load the string with the yaml module. If successful it's a yaml file.
+        """
+        return self._looks_like_yaml(file_prefix)
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'application/yaml'
+
+    def _looks_like_yaml(self, file_prefix):
+        # Pattern used by SequenceSplitLocations
+        if file_prefix.file_size < 50000 and not file_prefix.truncated:
+            # If the file is small enough - don't guess just check.
+            try:
+                item = yaml.load(file_prefix.contents_header)
+                assert isinstance(item, (list, dict))
+                return True
+            except Exception:
+                return False
+        else:
+            start = file_prefix.string_io().read(100).strip()
+            if start:
+                # If file is too big,
```

Comment:
We could be a little bit more explicit here: 
Read line by line and split after 10.000 lines or so on a line that doesn't start with space. You should be able to load the first chunk into the yaml parser, imho. There are some cases, like multi-line lists and so, but they are rare imho.